src/materialized/bin/materialized.rs

```diff
@@ -114,6 +114,11 @@ fn run() -> Result<(), failure::Error> {
     );
     opts.optopt("", "symbiosis", "(internal use only)", "URL");
     opts.optflag("", "no-prometheus", "Do not gather prometheus metrics");
+    opts.optflag(
+        "",
+        "debug",
+        "Allow running the debug (unoptimized) build. Do not use with the release build.",
```

Comment:
If we do stick with this design (see top-level comment), we may want to call this "dev", not "debug". Cargo's profiles are "dev" and "release" (kinda weird, I know, but those are the terms they use), and I think `--dev` reads a bit better than `--debug`; the latter reads as if you're asking materialized to enter some sort of special debugging mode.